packages/ic-management/src/types/ic-management.params.ts

```diff
@@ -23,13 +29,21 @@ export const toCanisterSettings = ({
   memoryAllocation,
   computeAllocation,
   reservedCyclesLimit,
+  logVisibility,
 }: CanisterSettings = {}): canister_settings => {
   return {
     controllers: toNullable(controllers?.map((c) => Principal.fromText(c))),
     freezing_threshold: toNullable(freezingThreshold),
     memory_allocation: toNullable(memoryAllocation),
     compute_allocation: toNullable(computeAllocation),
     reserved_cycles_limit: toNullable(reservedCyclesLimit),
+    log_visibility: isNullish(logVisibility)
+      ? []
+      : [
+          logVisibility === LogVisibility.Public
+            ? { public: null }
+            : { controllers: null },
```

Comment:
This could be fragile if another value is added to the enum.
Then it would still convert anything that's not `Public` to `controllers`.
How about throwing an error if the value is neither `Public` nor `Controllers`?